src/vcf/record.cpp

```diff
@@ -241,6 +243,45 @@ namespace ebi
                     break;
                 }
             }
+            
+            if (!found) {
+                std::vector<std::string> values;
+                util::string_split(field.second, ",", values);
+                if (source->version == Version::v41 || source->version == Version::v42) {
+                    for (auto & value : values)
+                        check_predefined_type("INFO", field.first, value, info_v41_v42);
+                    check_predefined_number("INFO", field.first, values.size(), info_v41_v42);
+                } else {
+                    for (auto & value : values)
+                        check_predefined_type("INFO", field.first, value, info_v43);
```

Comment:
Wouldn't it be possible to use the existing `check_field_type` and `check_field_cardinality` methods? The expected type or number would need to be extracted from the map before the method call, but otherwise I think they should be fine. That would also bring more consistency to the error reports and prevent code to be changed, for instance, in `check_field_type` but not in `check_predefined_type`.